astropy/io/fits/fitsrec.py

```diff
@@ -333,8 +333,7 @@ def from_columns(cls, columns, nrows=0, fill=False, character_as_bytes=False):
                     dim = arr.shape[0]
                 else:
                     dim = 0
-                if dim > nrows:
-                    nrows = dim
+                nrows = max(dim, nrows)
```

Comment:
Fine to just stick with the simple change here, but the whole thing could be
```
if nrows == 0 and (lengths := [arr.shape[0] for arr in columns._arrays if arr is not None]):
    nrows = max(lengths)
```
(Not completely sure the guard against all arrays being `None` is necessary)